### PR TITLE
lci: new package

### PR DESCRIPTION
**Pull request by JiakunYan**
Add a new package to the Spack built-in repo: lci.

The Lightweight Communication Interface (LCI) is designed to be an efficient communication library for multithreaded, irregular communications.

### CODE REVIEW

**Comment by JiakunYan** [comment]
@spackbot fix style

**Review by omor1 — APPROVED** [review]
lgtm

**Comment by JiakunYan** [comment]
Could someone help take a look at this PR?

File: var/spack/repos/builtin/packages/lci/package.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+        if not self.spec.satisfies("enable-pm=auto"):
+            arg = self.define("LCT_PMI_BACKEND_ENABLE_PMI1", "enable-pm=pmi1" in self.spec)
+            args.append(arg)
+            arg = self.define("LCT_PMI_BACKEND_ENABLE_PMI2", "enable-pm=pmi2" in self.spec)
+            args.append(arg)
+            arg = self.define("LCT_PMI_BACKEND_ENABLE_MPI", "enable-pm=mpi" in self.spec)
+            args.append(arg)
+            arg = self.define("LCT_PMI_BACKEND_ENABLE_PMIX", "enable-pm=pmix" in self.spec)
+            args.append(arg)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/lci/package.py]: Can you just use `args.extend` here and elsewhere + a literal list as an argument?
```python
args.extend([
    self.define("LCT_PMI_BACKEND_ENABLE_PMI1", self.spec.satisfies("enable-pm=pmi1")),
    ...
])
```

File: var/spack/repos/builtin/packages/lci/package.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    version("1.7.6", sha256="c88ccea2ad277ed38fc23187771b52b6fb212ed4429114717bfa8887ed21665c")
+    version("1.7.5", sha256="13e4084c9e7aaf55966ba5aa0423164b8fd21ee7526fc62017b3c9b3db99cb83")
+    version("1.7.4", sha256="00c6ef06bf90a02b55c72076dedf912580dcb1fb59fdc0e771d9e1a71283b72f")
+    version("1.7.3", sha256="3c47d51d4925e6700294ac060c88a73c26ca6e9df5b4010d0e90b0bf5e505040")
+
+    def is_positive_int(val):
+        try:
+            return int(val) > 0
+        except ValueError:
+            return val == "auto"
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/lci/package.py]: Can we define this outside of the class?

**Review by alalazo — COMMENTED** [review]
Some initial comments on style

**Comment by JiakunYan** [comment]
@alalazo I have resolved the comments. Could you take a look again?

**Comment by JiakunYan** [comment]
ping @alalazo @adamjstewart

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Going to force merge. This is the pipeline failure:
Screenshot from 2024-06-27 10-29-25

**Pull request merged**